### PR TITLE
feat(backup/s3): log start of all operations

### LINKED ISSUE #10218: S3 backup store is observable
- [ ] All operations provide useful logging output
- [ ] ~~AWS client metrics can be exported~~ The SDK only supports exporting to CloudWatch

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This just adds a single logging statement at the start of each method. Together with logging from the backup manager, this should provide us with all information we need.

Exporting metrics was not possible, the AWS SDK only supports exporting to CloudWatch. Providing custom exporters is possible but we'd have to convert from the SDK metric types to prometheus which is not trivial. I found no libraries which would do this for us.

closes #10218

### CODE REVIEW

**Review by deepthidevaki — APPROVED** [review]
Looks good. I'm not sure if all of them have to be at info level. But you can leave it as it is, and we will see eventually if they are useful or not.

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java [hunk 1]
```diff
@@ -101,6 +103,7 @@ public CompletableFuture<Void> save(final Backup backup) {
 
   @Override
   public CompletableFuture<BackupStatus> getStatus(final BackupIdentifier id) {
+    LOG.info("Querying status of {}", id);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java:106]: Let's leave it at info for now and adjust if it's too noisy. Currently there's a nice symmetry where all public methods from `BackupStore` are logging at info and all private implementation methods are logging at debug :)

**Comment by oleschoenburg** [comment]
I forgot to declare a dependency to `slf4j-api`.

bors r+

**Pull request merged**